types/src/main/scala/roc/types/decoders.scala

```diff
@@ -23,7 +26,7 @@ object decoders {
 
   implicit val stringElementDecoder: ElementDecoder[String] = new ElementDecoder[String] {
     def textDecoder(text: String): String         = text
-    def binaryDecoder(bytes: Array[Byte]): String = bytes.map(_.toChar).mkString
```

Comment:
Yeah, for binary decoders you pretty much have to look at the postgres source itself and port it from C to Scala.  These were built by porting the *recv functions from backend/utils/adt in the postgres source tree.